### DISCUSSION TITLE
stereo information not carried over after reading a SDF file

### DISCUSSION THREAD

**Discussion by jamiekl** [rdkit/rdkit#8114] [Q&A] [answered]
Hi,

I am new to RDKit. I have been working with structures with wedge bonds and noticed that when I read the SDF file using SDMolSupplier, somehow the information of wedge bond is not carried over. For eg:

```

  Mrv2403 12172409162D            

  4  3  0  0  0  0            999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  0  0  0  0
  2  3  1  1  0  0  0
  3  4  1  0  0  0  0
M  END
$$$$
```
the bond between 2 and 3 is a wedge bond. but when I read this SDF file, the wedge bond information is automatically ignored and I get the output
```

     RDKit          2D

  4  3  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  0
  2  3  1  0
  3  4  1  0
M  END
```
the code:
```python
from rdkit import Chem

sdf_file = "example.sdf"

load = Chem.SDMolSupplier(sdf_file)
for mol in load:
    print(Chem.MolToMolBlock(mol))
```    
 How to solve this?

**Comment by ptosco** [comment]
By default RDKit will strip all wedging information which it considers redundant or inappropriate (e.g., because the center it refers to is not chiral). Also, RDKit will recompute wedge positions based on internal rules, so wedges might not be on the same bonds where they were on the original block.
You may revert to the original wedging specified in the molblock with
```
from rdkit import Chem

sdf_file = """
  Mrv2403 12172409162D            

  4  3  0  0  0  0            999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  0  0  0  0
  2  3  1  1  0  0  0
  3  4  1  0  0  0  0
M  END
$$$$
"""

load = Chem.SDMolSupplier()
load.SetData(sdf_file)
for mol in load:
    Chem.ReapplyMolBlockWedging(mol)
    print(Chem.MolToMolBlock(mol))

     RDKit          2D

  4  3  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  0
  2  3  1  1
  3  4  1  0
M  END
```

**Reply by jamiekl** [reply]
Thank you for your reply. 

I also noticed that any stereo information from the input is stripped away. How can I control this? Does RDKit, by default, attempt to fix the stereo information when loading molecules using `SDMolSupplier`?

**Comment by ptosco** [comment]
Can you provide an example?

**Reply by jamiekl** [reply]
input 1: a molecule with a wiggly bond
```

  Mrv2403 12152423012D            

  4  3  0  0  0  0            999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  0  0  0  0
  2  3  1  4  0  0  0
  3  4  1  0  0  0  0
M  END
$$$$
```
output 1: the wiggly bond information is lost.
```  
 RDKit          2D

  4  3  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  0
  2  3  1  0
  3  4  1  0
M  END
$$$$
```
input 2: a molecule with straight double bond.  in this case, the RDKit automatically fixes the straight double bond which is supposedly having undefined stereo to a molecule with any stereo (cis or trans). I am curious to know how RDKit does it by default. 
```

  Mrv2403 12152421472D            

  8  7  0  0  0  0            999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.8579    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.6829    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.5079    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.3329    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  3  4  2  0  0  0  0
  4  5  1  0  0  0  0
  5  6  1  0  0  0  0
  6  7  2  0  0  0  0
  7  8  1  0  0  0  0
M  END
$$$$
```
output 2: 
```

     RDKit          2D

  8  7  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7145    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.4289    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.1434    0.4125    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.8579    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.6829    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.5079    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.3329    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  2  0
  4  5  1  0
  5  6  1  0
  6  7  2  3
  7  8  1  0
M  END
$$$$
```

**Reply by greglandrum** [reply]
@jamiekl a side comment: when you paste in mol or SDF data, please surround it with triple back quotes so that github displays it properly (with all the spaces).
I've edited your last comment to be correct and as an example

**Reply by greglandrum** [reply]
@jamiekl as Paolo said in his first answer: the RDKit removes information about sterochemistry that doesn't make sense. For example, in your first molecule:
![image](https://github.com/user-attachments/assets/9a97e330-a31d-4d89-ba0c-a673df51175e)
there is a wiggly bond from a double bond which cannot have cis/trans sterochemistry. The RDKit removes this.

If you have wiggly bond to a double bond that can have stereo:
```

     RDKit          2D

  5  4  0  0  0  0  0  0  0  0999 V2000
   -7.4242    4.9394    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -5.9242    4.9394    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -8.1742    6.2384    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -5.1742    3.6404    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -5.9242    2.3413    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  0
  1  3  1  0
  2  4  1  4
  4  5  1  0
M  END
```
The resulting mol block indicates that the double bond has unknown stereo by writing it as a crossed bond:
```

     RDKit          2D

  5  4  0  0  0  0  0  0  0  0999 V2000
   -7.4242    4.9394    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -5.9242    4.9394    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -8.1742    6.2384    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -5.1742    3.6404    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -5.9242    2.3413    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  2  3
  1  3  1  0
  2  4  1  0
  4  5  1  0
M  END
```

**Reply by greglandrum** [reply, marked as answer]
With regard to the second part of your question, about this molecule:
![image](https://github.com/user-attachments/assets/498fffc5-3cae-4ae6-8664-880dbc5a1a5e)
(the crossed double bond was not in your input, the RDKit adds this when drawing unspecified double bonds)

The RDKit recognizes that the linear double bond is intended to indicate unknown stereochemistry and writes the output accordingly (again, with an explicitly crossed double bond)

**Reply by jamiekl** [reply]
@greglandrum Thanks for the suggestions and I can presume that RDKit fixes this automatically when encountering unknown stereochemistry. is there a way to control RDKit from fixing it automatically?

**Reply by greglandrum** [reply]
> @greglandrum Thanks for the suggestions and I can presume that RDKit fixes this automatically when encountering unknown stereochemistry. is there a way to control RDKit from fixing it automatically?

I can interpret that a lot of different ways. What exactly do you want to happen?

**Reply by jamiekl** [reply]
I want to inspect and identify if there are unknown stereochemistry without it being automatically resolved by RDKit.
